src/Features/CSharp/Portable/RemoveUnusedParametersAndValues/CSharpRemoveUnusedValuesCodeFixProvider.cs

```diff
@@ -0,0 +1,235 @@
+                // the replacement statement list tracked in _statementsBuilder.
+
+                if (_statementsBuilder.Count == 0)
+                {
+                    // Nothing to replace.
+                    return;
+                }
+
+                // Move the leading trivia from original local declaration statement
+                // to the first statement of the replacement statement list.
+                var leadingTrivia = _localDeclarationStatement.Declaration.Type.GetLeadingTrivia()
+                    .Concat(_localDeclarationStatement.Declaration.Type.GetTrailingTrivia());
+                _statementsBuilder[0] = _statementsBuilder[0].WithLeadingTrivia(leadingTrivia);
+
+                // Move the trailing trivia from original local declaration statement
+                // to the last statement of the replacement statement list.
+                var last = _statementsBuilder.Count - 1;
+                var trailingTrivia = _localDeclarationStatement.SemicolonToken.GetAllTrivia();
+                _statementsBuilder[last] = _statementsBuilder[last].WithTrailingTrivia(trailingTrivia);
+
+                // Replace the original local declaration statement with new statement list
+                // from _statementsBuilder.
+                if (_localDeclarationStatement.Parent is BlockSyntax)
+                {
+                    _editor.InsertAfter(_localDeclarationStatement, _statementsBuilder.Skip(1));
+                    _editor.ReplaceNode(_localDeclarationStatement, _statementsBuilder[0]);
+                }
+                else
+                {
+                    _editor.ReplaceNode(_localDeclarationStatement, SyntaxFactory.Block(_statementsBuilder));
+                }
+            }
+
+            private void GenerateDeclarationStatementForCurrentNonDiscardVariables()
+            {
+                // Convert a discard declaration with initializer of the form "var _ = M();" into
+                // a discard assignment "_ = M();"
+                if (_currentNonDiscardVariables.Count > 0)
+                {
+                    var statement = SyntaxFactory.LocalDeclarationStatement(
+                                        SyntaxFactory.VariableDeclaration(_localDeclarationStatement.Declaration.Type, _currentNonDiscardVariables))
+                                    .WithAdditionalAnnotations(Formatter.Annotation);
+                    _statementsBuilder.Add(statement);
+                    _currentNonDiscardVariables = new SeparatedSyntaxList<VariableDeclaratorSyntax>();
+                }
+            }
+
+            private void GenerateAssignmentForDiscardVariable(VariableDeclaratorSyntax variable)
+            {
+                Debug.Assert(IsDiscardDeclaration(variable));
+
+                // Convert a discard declaration with initializer of the form "var _ = M();" into
+                // a discard assignment "_ = M();"
```

Comment:
this comment seems accurate.  i think the commit in the method above is incorrect.

Reply:
Yes.

---
In reply to: [234800001](https://github.com/dotnet/roslyn/pull/30777#discussion_r234800001) [](ancestors = 234800001)